lib/trie/proof_test.go

```diff
@@ -0,0 +1,143 @@
+package trie
+
+import (
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"testing"
+
+	crand "crypto/rand"
+
+	"github.com/ChainSafe/chaindb"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/stretchr/testify/require"
+)
+
+func inMemoryChainDB(t *testing.T) (*chaindb.BadgerDB, func()) {
+	t.Helper()
+
+	tmpdir, err := ioutil.TempDir("", "trie-chaindb-*")
+	require.NoError(t, err)
+
+	db, err := chaindb.NewBadgerDB(&chaindb.Config{
+		InMemory: true,
+		DataDir:  tmpdir,
+	})
+	require.NoError(t, err)
+
+	clear := func() {
+		err := db.ClearAll()
```

Comment:
No need to `ClearAll`. You can directly `Close` and `RemoveAll`

Reply:
Done